test/integration/groups.js

```diff
@@ -620,3 +620,137 @@ test('group members replicate each other eventually', async (t) => {
     p(carol.close)(true),
   ])
 })
+
+test('Group is still intact after excluding someone', async (t) => {
+  const sharedConfig = {
+    friends: {
+      hops: 1,
+    },
+    replicationScheduler: {
+      debouncePeriod: 1,
+      partialReplication: {
+        0: [{}],
+        1: [{ purpose: 'main' }, { purpose: 'group/additions' }],
+        group: [{ purpose: '$groupSecret' }],
+      },
+    },
+  }
+  const alice = Server('alice', {
+    ...sharedConfig,
+    metafeeds: {
+      seed: aliceSeed,
+    },
+  })
+
+  const bob = Server('bob', {
+    ...sharedConfig,
+    metafeeds: {
+      seed: bobSeed,
+    },
+  })
+
+  const carol = Server('carol', {
+    ...sharedConfig,
+    metafeeds: {
+      seed: carolSeed,
+    },
+  })
+
+  await alice.tribes2.start()
+  await bob.tribes2.start()
+  await carol.tribes2.start()
+
+  await p(alice.metafeeds.findOrCreate)()
+  const bobRoot = await p(bob.metafeeds.findOrCreate)()
+  const carolRoot = await p(carol.metafeeds.findOrCreate)()
+  t.pass('created root metafeeds for alice, bob, carol')
+
+  await Promise.all([
+    p(alice.db.publish)(u.follow(bob.id)),
+    p(alice.db.publish)(u.follow(carol.id)),
+    p(bob.db.publish)(u.follow(alice.id)),
+    p(carol.db.publish)(u.follow(alice.id)),
+  ])
+  t.pass('alice is mutually following bob and carol')
+
+  const connectionBA1 = await p(bob.connect)(alice.getAddress())
+  const connectionCA1 = await p(carol.connect)(alice.getAddress())
+  t.pass('bob and carol connected to alice')
+  await sleep(REPLICATION_TIMEOUT)
+  await sleep(REPLICATION_TIMEOUT)
```

Comment:
little bit brittle. I suspect we can copy the `replicate` function from ssb-tribes2, or look at extracting that into `scuttle-testbot` as a nice helper many can use

Reply:
oh ... I mean we would want the "isSync" part, where we await till it's done replicating.